### ISSUE TITLE
Sync with a LDAP server

### ISSUE THREAD

**Issue by aaronduino** [monicahq/monica#1539]
**Is your feature request related to a problem? Please describe.**
It is difficult to keep information up-to-date and consistent between Monica and Google Contacts, the latter of which syncs information between my laptop and phone.

**Describe the solution you'd like**
Monica could add support for syncing vCard info with a LDAP server of the user's choice (I am not sure if I am using the right terminology but hopefully what I am saying makes sense). In my case, I'd sync it with Google Contacts' LDAP interface; that way, contact info would always be consistent and up-to-date between all my devices and Monica.

**Describe alternatives you've considered (optional)**
- Creating a script to sync between a Monica API and a LDAP server.
  This does not sound like a clean/stable/efficient solution, but it _could_ work.

**Additional context**
Hopefully, letting the user sync with a LDAP server would put the data more in control of the user. Additionally, it would act as an easy backup solution.

Thanks for Monica!

**Comment by ChucklesTheBeard** [comment]
Possible solution to #125

**Comment by aaronduino** [comment]
Correction:

I think I got the terminology wrong.

What we probably want instead is CardDAV, a standard that is supported by google contacts.

https://developers.google.com/contacts/carddav/

**Comment by ChucklesTheBeard** [comment]
In that case, duplicate of #42 and #582, and being worked on in #1284

**Comment by aaronduino** [comment]
Sweet, good to hear. In that case, I'll close this issue

**Comment by anmol26s** [comment]
A Ldap support for Monica to connect users from Yunohost server would be greatly appreciated.

**Comment by thies226j** [comment]
Yeah LDAP support would still be a great integration. As an address book on the one hand and as an auth provider on the other hand.

**Comment by hutch0** [comment]
I would love to see LDAP support as an address book backend. I host a bunch of services on my servers with LDAP authentication and SSO. It would be awesome to have the workflow go from meeting someone and putting their info in my mobile and having that sync over CardDAV to Monica, then getting to know them better and then adding them to the services group in LDAP, which would automatically enable everything for them. 
I'd be happy to help with it. I haven't written any PHP, mostly Perl and Python, but there's a first time for everything if someone can point me in the right direction.

**Comment by curiousercreative** [comment]
> I would love to see LDAP support as an address book backend. I host a bunch of services on my servers with LDAP authentication and SSO. It would be awesome to have the workflow go from meeting someone and putting their info in my mobile and having that sync over CardDAV to Monica, then getting to know them better and then adding them to the services group in LDAP, which would automatically enable everything for them. I'd be happy to help with it. I haven't written any PHP, mostly Perl and Python, but there's a first time for everything if someone can point me in the right direction.

@hutch0 I share your dream, were you able to realize the dream?
